openmc/settings.py

```diff
@@ -1275,6 +1281,12 @@ def _keff_trigger_from_xml_element(self, root):
     def _source_from_xml_element(self, root):
         for elem in root.findall('source'):
             self.source.append(Source.from_xml_element(elem))
+            if isinstance(Source.from_xml_element(elem), MeshIndependent):
+                mesh_id = int(get_text(elem, 'mesh'))
+                path = f"./mesh[@id='{mesh_id}']"
+                mesh_elem = root.find(path)
+                if mesh_elem is not None:
+                    self.source.space.mesh = MeshBase.from_xml_element(mesh_elem)
```

Comment:
Let's read in the source object and add it to the `Source.source` attribute at the very end to simplify the code a bit.
```suggestion
        for elem in root.findall('source'):
            src = Source.from_xml_element(elem)
            if isinstance(src.space, MeshIndependent):
                mesh_id = int(get_text(elem, 'mesh'))
                path = f"./mesh[@id='{mesh_id}']"
                mesh_elem = root.find(path)
                if mesh_elem is not None:
                    src.space.mesh = MeshBase.from_xml_element(mesh_elem)
            self.source.append(src)
```
This also avoids an additional read of the source element in the if-statement.

Reply:
Let's raise a `RuntimeError` if the mesh isn't found in the `settings.xml` file as well. We'll have no way to set that mesh correctly after this point so we should make sure the source object isn't hanging around without it.